types/block.go

```diff
@@ -988,29 +1002,127 @@ func CommitFromProto(cp *tmproto.Commit) (*Commit, error) {
 
 //-----------------------------------------------------------------------------
 
-// Data contains the set of transactions included in the block
+// Data contains all the available Data of the block.
+// Data with reserved namespaces (Txs, IntermediateStateRoots, Evidence) and
+// Celestia application specific Messages.
 type Data struct {
-
 	// Txs that will be applied by state @ block.Height+1.
 	// NOTE: not all txs here are valid.  We're just agreeing on the order first.
 	// This means that block.AppHash does not include these txs.
 	Txs Txs `json:"txs"`
 
+	Evidence EvidenceData `json:"evidence"`
+
+	// The messages included in this block.
+	// TODO: how do messages end up here? (abci) app <-> ll-core?
+	// A simple approach could be: include them in the Tx above and
+	// have a mechanism to split them out somehow? Probably better to include
+	// them only when necessary (before proposing the block) as messages do not
+	// really need to be processed by tendermint
+	Messages Messages `json:"msgs"`
+
+	// OriginalSquareSize is the size of the square after splitting all the block data
+	// into shares. The erasure data is discarded after generation, and keeping this
+	// value avoids unnecessarily regenerating all of the shares when returning
+	// proofs that some element was included in the block
+	OriginalSquareSize uint64 `json:"square_size"`
+
 	// Volatile
 	hash tmbytes.HexBytes
 }
 
-// Hash returns the hash of the data
+// Hash returns the hash of the data. `data.hash` is expected to be set by
+// PrepareProposal in celestia-app. However, this function falls back to
+// calculating `data.hash` as vanilla tendermint would in order to satisfy unit
+// and e2e tests.
```

Comment:
Similar comment, if this is now here exclusively not to break the tests, would it make sense to move out of production code?